Wallet/Cryptography/KeyVault.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Wallet.Cryptography
 {
-    public class KeyVault : IKeyVault
+    public class KeyVault : IDatabase
```

Comment:
find a more generic name for the interface.
the implementations will be database/sql, keyvault and redis

Reply:
ISecretsStore? @Ilanak I will change in my next iteration...